### ISSUE TITLE
`aws_opsworks_permission` fails to update own permissions

### ISSUE THREAD

**Issue by ghost** [terraform-providers/terraform-provider-aws#4804]
When using `aws_opsworks_permission` to **update** (read: create works) the opsworks permissions of your own user it fails with _"ValidationException: Unable to change own permission level. Only AllowSsh and AllowSudo allowed."_ even when `level` is unset.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.7
+ provider.aws v1.22.0
+ provider.template v1.0.0
```

### Affected Resource(s)

* `aws_opsworks_permission`

#### First run:

Where the `opsworks_permission_user_arn` variable is a list that contains your iam_arn and at least one other to prove the issue only applies to when making changes to your own iam user.

```hcl
variable "opsworks_permission_user_arn" {
  type        = "list"
  description = "A list of user arns that should be given access to the stack."
}

resource "aws_opsworks_permission" "stack_permission" {
  count      = "${length(var.opsworks_permission_user_arn)}"
  allow_ssh  = true
  allow_sudo = true
  user_arn   = "${var.opsworks_permission_user_arn[count.index]}"
  stack_id   = "${aws_opsworks_stack.main.id}"
}
```

#### Second run:

Where the `opsworks_permission_user_arn` variable is a list that contains your iam_arn and at least one other to prove the issue only applies to when making changes to your own iam user.

```hcl
variable "opsworks_permission_user_arn" {
  type        = "list"
  description = "A list of user arns that should be given access to the stack."
}

resource "aws_opsworks_permission" "stack_permission" {
  count      = "${length(var.opsworks_permission_user_arn)}"
  allow_ssh  = true
  allow_sudo = false
  user_arn   = "${var.opsworks_permission_user_arn[count.index]}"
  stack_id   = "${aws_opsworks_stack.main.id}"
}
```

### Expected Behavior

All users should now be allowed to SSH in (`allow_ssh = true`) but shouldn't have sudo permissions ('allow_sudo = false').

### Actual Behavior

All users in the list will have had their sudo permissions revoked **except** your own IAM user.

### Steps to Reproduce

1. `terraform apply`
2. Update `allow_sudo` from `true` to `false`.
3. `terraform apply`

### References

* #458 - similar issue, reporting that `aws_opsworks_permission` doesn't work at all when applied to your own IAM user.
* #1038 - fixes #458, so `aws_opsworks_permission` applies to your own user.

**Comment by bflad** [comment]
Bug fix submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/11379

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.45.0 of the Terraform AWS Provider, Thursday next week.
